openforcefield/tests/conftest.py

```diff
@@ -66,18 +70,46 @@ def pytest_addoption(parser):
     If --runslow is not given, tests marked with pytest.mark.slow are
     skipped.
 
-    If --failwip is not give, tests marked with pytest.mark.wip are
+    If --failwip is not given, tests marked with pytest.mark.wip are
     xfailed.
+
+    Parameters
+    ----------
+    parser : argparsing.parser
+        The parser used by pytest to process arguments
+
+    Returns
+    -------
+    None
+
     """
-    parser.addoption(
-        "--runslow", action="store_true", default=False, help="run slow tests"
-    )
-    parser.addoption(
-        "--failwip",
-        action="store_true",
-        default=False,
-        help="fail work in progress tests",
-    )
+
+    # Loaded pytest plugins define their own arguments, and in certain cases
+    # our options use the same name. Although this can define two different
+    # behaviors for the same argument, the two options below are fairly
+    # unambiguous in their interpretation, so we skip any errors and allow
+    # the testing to proceed
+
+    try:
+        parser.addoption(
+            "--runslow", action="store_true", default=False, help="run slow tests"
+        )
+    except ValueError:
+        logger.warning(
+            "Option --runslow already added elsewhere (from a plugin possibly?). Skipping..."
+        )
+
```

Comment:
Would it be possible, in the case of `--runslow` already being registered, to register _our_ `--runslow` as something with a more unique name, like `--runslowopenff`?

Reply:
Then we expose a CLI where both `--runslow` and `--runslowopenff` are set and enabled. I think this proceeds down a tangled path and would not be very clean. If we look at the other side, should qcfractal change theirs to `--runslowqcfractal` if openff is installed?

In this case, `runslow` is a boolean that means the same thing everywhere. Do we want to run tests we consider slow? Yes or no; there is no other thing this option can mean. This means if someone else defines it, no problem, we already know what to do with it.

(Same thing with --failwip)

On a related tangent, it is aggravating that pytest actually has this behavior. Nearly every other CLI I encounter uses a "last definition wins" behavior, which is fairly sane and useful when e.g. overriding system-defined defaults.

Reply:
> 
Then we expose a CLI where both --runslow and --runslowopenff are set and enabled.

I'd personally be in support of just renaming _our_ `--runslow` to something that only our software will call, and only ever call `--runslowopenff`. This wouldn't resolve the issue of QCFractal's argument colliding with other packages that a dev or bot may have installed, but it would fix our issue without requiring changes from others. A downside would be the command in the toolkit colliding with the command in another OpenFF package.

I agree with your last point, it's almost surprising that this is the case. I haven't looked into any workarounds, but it seems unlikely we're the first people to run into it.

Reply:
Both seem like good options, but I'd be most in favor of eventually changing ours to `--runslowopenff` (or something even more specific like `runslowopenfftoolkit`). I could see a strange-but-not-impossible case arising where some other package has their `--runslow` kwarg do something other than `action=store_true`, and then this fix gives us a weird, untraceable behavior.